src/muse/demand_share.py

```diff
@@ -474,28 +471,30 @@ def _inner_split(
     """
     from numpy import logical_and
 
+    # Find decrease in capacity production by each asset over time
     shares: Mapping[Hashable, xr.DataArray] = {
         key: method(capacity=capacity)
         .groupby("technology")
         .sum("asset")
         .rename(technology="asset")
         for key, capacity in assets.items()
     }
+
+    # Total decrease in production across assets
     try:
         summed_shares: xr.DataArray = xr.concat(shares.values(), dim="concat_dim").sum(
             "concat_dim"
         )
-
-        # Calculates the total demand assigned in the previous step with the "method"
-        # function across agents and assets.
         total: xr.DataArray = summed_shares.sum("asset")
     except AttributeError:
         raise AgentWithNoAssetsInDemandShare()
 
     # Calculates the demand divided by the number of assets times the number of agents
     # if the demand is bigger than zero and the total demand assigned with the "method"
     # function is zero.
-    unassigned = (demand / (len(shares) * len(summed_shares))).where(
```

Comment:
Reeeeeally bad idea to call `len` on a DataArray (`summed_shares`), as this will just take the length of the first dimension, which isn't guaranteed to be the dimension you're after ("asset" in this case), unless you're really careful about it. In this case we get away with it, but this has bitten me on another PR and took me a long time to figure out.

Also, I still don't really understand what this function is doing. Why is it multiplying the number of agents by the number of assets? What does `totals` represent?